### PR TITLE
[0.6.2] Enhance richness of telemetry

### PR DESCRIPTION
**Pull request by ingalless**
This PR includes the following:
* Configurable automatic anonymous user tracking
* Authenticated user tracking
* Forwarding all `Log` calls to app insights as trace messages
* Associating exceptions with their requests
* Custom event when no migrations are ran
* Improved readme

### CODE REVIEW

**Review by danielyates-mondago — APPROVED** [review]
Good work 👍 

I've left comments, but these shouldn't block it from being merged.

File: config/application_insights.php [hunk 1]
```diff
@@ -6,7 +6,12 @@
      */
     'instrumentation_key' => env('APPINSIGHTS_INSTRUMENTATIONKEY'),
     /*
-     * Microsoft Application Insights instrumentation key.
+     * Whether to send data
+     */
```
[the remark this answers is not in the document]
Comment (ingalless) [reply config/application_insights.php:10]: I like it from a "plug and play" perspective. If you include the package you're likely to want to send the data. We then set the environment variable to false in our local .env file. Does that make sense? Happy to make the change :)

[the remark this answers is not in the document]
Comment (danielyates-mondago) [reply config/application_insights.php:10]: It does makes sense. Would it cause any problems if the `APPINSIGHTS_INSTRUMENTATIONKEY` wasn't set?

File: src/Logging/ApplicationInsightsHandler.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+namespace Mondago\ApplicationInsights\Logging;
+
+use ApplicationInsights\Channel\Contracts\Message_Severity_Level;
+use Mondago\ApplicationInsights\ApplicationInsights;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Logger;
+
+class ApplicationInsightsHandler extends AbstractProcessingHandler
+{
+    const InsightToLoggingInterfaceMapping = [
```
[the remark this answers is not in the document]
Comment (ingalless) [reply src/Logging/ApplicationInsightsHandler.php:12]: The match would look like this, wdyt?

```php
    protected function write(array $record): void
    {
        $level = match($record['level']) {
            Logger::EMERGENCY => Message_Severity_Level::CRITICAL,
            Logger::ALERT     => Message_Severity_Level::CRITICAL,
            Logger::CRITICAL  => Message_Severity_Level::CRITICAL,
            Logger::ERROR     => Message_Severity_Level::ERROR,
            Logger::WARNING   => Message_Severity_Level::WARNING,
            Logger::NOTICE    => Message_Severity_Level::INFORMATION,
            Logger::INFO      => Message_Severity_Level::INFORMATION,
            Logger::DEBUG     => Message_Severity_Level::VERBOSE,
        };

        if (isset($record['context']['exception'])) {
            $this->client->trackException($record['context']['exception']);
        } else {
            $this->client->trackMessage(
                (string) $record['message'],
                $level,
                $record['context'],
            );
        }
    }
```

I personally prefer it as a property.

[the remark this answers is not in the document]
Comment (danielyates-mondago) [reply src/Logging/ApplicationInsightsHandler.php:12]: Looking at it there isn't really much in it, I'd say it's probably not worth doing. The only benefit you'd get is being able to use a default case.

File: config/application_insights.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ingalless) [reply config/application_insights.php:10]: Exceptions silently throw, and bubble up to the next handler, so you shouldn't have any problems if you have the combo of enabled and no instrumentationkey

**Pull request merged**